dll/win32/shell32/systray.cpp

```diff
@@ -45,6 +48,7 @@ BOOL WINAPI Shell_NotifyIconA(DWORD dwMessage, PNOTIFYICONDATAA pnid)
     /* Validate the structure size and the flags */
     cbSize = pnid->cbSize;
     dwValidFlags = NIF_MESSAGE | NIF_ICON | NIF_TIP;
+    static_assert(sizeof(*pnid) >= NOTIFYICONDATAA_V3_SIZE, "V3 or later is needed");
```

Comment:
Since this appears to be required in all this file and the CUserNotification.cpp one, I wonder whether a static_assert done globally in those two files (or even at the level of the whole shell32) wouldn't be better?

Reply:
Should we move `#include "compat.h"` into `"precomp.h"`?

Reply:
Should we move `#include "compat.h"` into `"precomp.h"`?

Reply:
Serge suggested basically to not use NOTIFYICONDATAA_V3_SIZE in systray.cpp at all. Not in any assert, and not within productive code.

Reply:
Without `NOTIFYICONDATAA_V3_SIZE`, we cannot  support Vista build.

Reply:
Why not?

Reply:
Because Vista code uses `NOTIFYICONDATAA_V3_SIZE`.